Add details to overlay algorithms (#4519)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/qgis/vectoroverlay.rst

     - Layer to extract (parts of) features from.
   * - **Overlay layer**
     - ``OVERLAY``
     - [vector: any]
     
     - Layer containing the geometries that will be subtracted from the

>        input layer geometries. It's expected to have at least as many

Reviewer: ```suggestion
       input layer geometries. It is expected to have at least as many
```

>        dimensions as the input layer geometries.

Reviewer: ```suggestion
       dimensions (point: 0D, line: 1D, polygon: 2D, volume: 3D) as the input
       layer geometries.
```

     - ``OVERLAY``
     - [vector: any]
     - Layer containing the features to check for overlap.
       Its features' geometry is expected to have at least as many

>        dimensions as the input layer's.

Reviewer: ```suggestion
       dimensions (point: 0D, line: 1D, polygon: 2D, volume: 3D)
       as the input layer's.
```
